# **Amazon E-commerce Sales and Fulfillment Performance Analysis** 


## Project Overview

### Domain: 
E-commerce Sales and Supply Chain Analytics

### Objective:
To analyze Amazon E-commerce sales data in order to understand Sales performance,Product demand,fulfillment efficiency, and shipping patterns,and to identify factors that influence revenue and order success.

### Business Problem: 
With the rapid growth of e-commerce, Amazon processes thousands of orders daily across different product categories and regions. However, raw sales and order data alone do not provide clear insights into which products perform best, how efficient the fulfillment process is, or where major customers are located. Without proper data analysis, it becomes difficult for business managers to make informed decisions on inventory planning, promotion strategies, and logistics optimization.

Therefore, this project aims to analyze Amazon e-commerce sales data to identify sales trends, high-demand products, regional sales distribution, and fulfillment performance, helping improve business decision-making and operational efficiency.

### Dataset:
Amazon sales data set containing 128975 rows and 23 features.It contains detailed information about customer orders, product details, sales values, shipping locations, and fulfillment status. This dataset helps analyze sales performance, customer demand, and logistics efficiency in an online retail environment.

### Step1: 
Data Loading and Initial Overview

### Goal:
Import the dataset, examining its structure and understand the basic characteristics of the data.

In [1]:
#Importing Pandas Libraries
import pandas as pd
import numpy as np


## Data Source and Loading

### Data Source: Kaggle- Amazon E-Commerce Sales Dataset.

### Source Link:
https://www.kaggle.com/code/harshilt512/amazon-e-commerce-sales-dataset/input

### Loading objective:
We will load the csv file and perform initial inspection to understand. Total number of records(rows) Total number of features(Columns) Column names and datatypes Overall data structure and quality

In [2]:
#Loading of Dataset
df=pd.read_csv(r"C:\Users\ME CONSTRUCTIONS\Amazon Sales.csv")
print("Dataset loaded successfully")

Dataset loaded successfully


### Preview of the Dataset

We will examine the first few rows of the data, 
* To verify data loaded correctly To view the structure of data.
*  To understand the type of data.
* To spot obvious errors early


The df.head() function displays the first 5 rows by default giving quick preview of the dataset.

In [3]:
#Display the first 10 rows of the dataset
print("First 10 rows of the dataset:")
df.head(10)

First 10 rows of the dataset:


index             Order ID      Date                        Status  \
0      0  405-8078784-5731545  04-30-22                     Cancelled   
1      1  171-9198151-1101146  04-30-22  Shipped - Delivered to Buyer   
2      2  404-0687676-7273146  04-30-22                       Shipped   
3      3  403-9615377-8133951  04-30-22                     Cancelled   
4      4  407-1069790-7240320  04-30-22                       Shipped   
5      5  404-1490984-4578765  04-30-22                       Shipped   
6      6  408-5748499-6859555  04-30-22                       Shipped   
7      7  406-7807733-3785945  04-30-22  Shipped - Delivered to Buyer   
8      8  407-5443024-5233168  04-30-22                     Cancelled   
9      9  402-4393761-0311520  04-30-22                       Shipped   

  Fulfilment Sales Channel  ship-service-level    Style                  SKU  \
0   Merchant      Amazon.in           Standard   SET389       SET389-KR-NP-S   
1   Merchant      Amazon.in           Standard  JNE3781      JNE3781-KR-XXXL   
2     Amazon      Amazon.in          Expedited  JNE3371        JNE3371-KR-XL   
3   Merchant      Amazon.in           Standard    J0341           J0341-DR-L   
4     Amazon      Amazon.in          Expedited  JNE3671      JNE3671-TU-XXXL   
5     Amazon      Amazon.in          Expedited   SET264      SET264-KR-NP-XL   
6     Amazon      Amazon.in          Expedited    J0095          J0095-SET-L   
7   Merchant      Amazon.in           Standard  JNE3405         JNE3405-KR-S   
8     Amazon      Amazon.in          Expedited   SET200  SET200-KR-NP-A-XXXL   
9     Amazon      Amazon.in          Expedited  JNE3461       JNE3461-KR-XXL   

        Category  ... Qty currency  Amount    ship-city     ship-state  \
0            Set  ...   0      INR  647.62       MUMBAI    MAHARASHTRA   
1          kurta  ...   1      INR  406.00    BENGALURU      KARNATAKA   
2          kurta  ...   1      INR  329.00  NAVI MUMBAI    MAHARASHTRA   
3  Western Dress  ...   0      INR  753.33   PUDUCHERRY     PUDUCHERRY   
4            Top  ...   1      INR  574.00      CHENNAI     TAMIL NADU   
5            Set  ...   1      INR  824.00    GHAZIABAD  UTTAR PRADESH   
6            Set  ...   1      INR  653.00   CHANDIGARH     CHANDIGARH   
7          kurta  ...   1      INR  399.00    HYDERABAD      TELANGANA   
8            Set  ...   0      NaN     NaN    HYDERABAD      TELANGANA   
9          kurta  ...   1      INR  363.00      Chennai     TAMIL NADU   

   ship-postal-code ship-country  \
0          400081.0           IN   
1          560085.0           IN   
2          410210.0           IN   
3          605008.0           IN   
4          600073.0           IN   
5          201102.0           IN   
6          160036.0           IN   
7          500032.0           IN   
8          500008.0           IN   
9          600041.0           IN   

                                       promotion-ids    B2B fulfilled-by  
0                                                NaN  False    Easy Ship  
1  Amazon PLCC Free-Financing Universal Merchant ...  False    Easy Ship  
2       IN Core Free Shipping 2015/04/08 23-48-5-108   True          NaN  
3                                                NaN  False    Easy Ship  
4                                                NaN  False          NaN  
5       IN Core Free Shipping 2015/04/08 23-48-5-108  False          NaN  
6       IN Core Free Shipping 2015/04/08 23-48-5-108  False          NaN  
7  Amazon PLCC Free-Financing Universal Merchant ...  False    Easy Ship  
8       IN Core Free Shipping 2015/04/08 23-48-5-108  False          NaN  
9                                                NaN  False          NaN  

[10 rows x 23 columns]

### Last rows of the dataset

Checking last rows of the data set help us to, verify the entire data loaded successfully Check if data patterns change at the end Ensure there is no error at the file`s end

df.tail(n), displays last n rows.

In [11]:
#To display last 10 rows of the dataset.

print("Last 10 rows of dataset")
df.tail()

Last 10 rows of dataset


index             Order ID       Date   Status Fulfilment  \
128970  128970  406-6001380-7673107 2022-05-31  Shipped     Amazon   
128971  128971  402-9551604-7544318 2022-05-31  Shipped     Amazon   
128972  128972  407-9547469-3152358 2022-05-31  Shipped     Amazon   
128973  128973  402-6184140-0545956 2022-05-31  Shipped     Amazon   
128974  128974  408-7436540-8728312 2022-05-31  Shipped     Amazon   

       Sales Channel  ship-service-level    Style             SKU  \
128970      Amazon.in          Expedited  JNE3697   JNE3697-KR-XL   
128971      Amazon.in          Expedited   SET401  SET401-KR-NP-M   
128972      Amazon.in          Expedited    J0157    J0157-DR-XXL   
128973      Amazon.in          Expedited    J0012    J0012-SKD-XS   
128974      Amazon.in          Expedited    J0003     J0003-SET-S   

             Category  ... Qty currency  Amount  ship-city    ship-state  \
128970          kurta  ...   1      INR   517.0  HYDERABAD     TELANGANA   
128971            Set  ...   1      INR   999.0   GURUGRAM       HARYANA   
128972  Western Dress  ...   1      INR   690.0  HYDERABAD     TELANGANA   
128973            Set  ...   1      INR  1199.0      Halol       Gujarat   
128974            Set  ...   1      INR   696.0     Raipur  CHHATTISGARH   

        ship-postal-code ship-country  \
128970          500013.0           IN   
128971          122004.0           IN   
128972          500049.0           IN   
128973          389350.0           IN   
128974          492014.0           IN   

                                       promotion-ids    B2B fulfilled-by  
128970                                           NaN  False          NaN  
128971  IN Core Free Shipping 2015/04/08 23-48-5-108  False          NaN  
128972                                           NaN  False          NaN  
128973  IN Core Free Shipping 2015/04/08 23-48-5-108  False          NaN  
128974  IN Core Free Shipping 2015/04/08 23-48-5-108  False          NaN  

[5 rows x 23 columns]

In [4]:
df.isnull().sum()

index                     0
Order ID                  0
Date                      0
Status                    0
Fulfilment                0
Sales Channel             0
ship-service-level        0
Style                     0
SKU                       0
Category                  0
Size                      0
ASIN                      0
Courier Status         6872
Qty                       0
currency               7795
Amount                 7795
ship-city                33
ship-state               33
ship-postal-code         33
ship-country             33
promotion-ids         49153
B2B                       0
fulfilled-by          89698
dtype: int64

In [5]:
df.dtypes

index                   int64
Order ID               object
Date                   object
Status                 object
Fulfilment             object
Sales Channel          object
ship-service-level     object
Style                  object
SKU                    object
Category               object
Size                   object
ASIN                   object
Courier Status         object
Qty                     int64
currency               object
Amount                float64
ship-city              object
ship-state             object
ship-postal-code      float64
ship-country           object
promotion-ids          object
B2B                      bool
fulfilled-by           object
dtype: object

In [7]:
#datatype conversion
df['Date']=pd.to_datetime(df['Date'],errors='coerce')

In [8]:
print(df['Date'].dtype)

datetime64[ns]


In [9]:
df['Date'].isnull().sum()

np.int64(0)

### Number of rows and columns

df.shape function help us to display number of rows(records) and columns(features)



In [10]:
#Number of rows and columns

df.shape

(128975, 23)

## information and Data Types

### Understanding data types is crucial for,
* Accurate analysis
* Proper data cleaning
* Correct processing of Dataset.

### The info() method provides:
* Total number of entries.
* Column names and data types.
* Number of non null values
* Memory usage

In [11]:
#Display dataset information

print("Dataset Information")
df.info()

Dataset Information
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 128975 entries, 0 to 128974
Data columns (total 23 columns):
 #   Column              Non-Null Count   Dtype         
---  ------              --------------   -----         
 0   index               128975 non-null  int64         
 1   Order ID            128975 non-null  object        
 2   Date                128975 non-null  datetime64[ns]
 3   Status              128975 non-null  object        
 4   Fulfilment          128975 non-null  object        
 5   Sales Channel       128975 non-null  object        
 6   ship-service-level  128975 non-null  object        
 7   Style               128975 non-null  object        
 8   SKU                 128975 non-null  object        
 9   Category            128975 non-null  object        
 10  Size                128975 non-null  object        
 11  ASIN                128975 non-null  object        
 12  Courier Status      122103 non-null  object        
 13  Qty      

### Column names  and Description

This dataset provides detailed insights into Amazon sales data, including SKU Code, Design Number, Stock, Category, Size and Color, to help optimize product profitability

* Category: Type of product. (String)
* Size: Size of the product. (String)
* Date: Date of the sale. (Date)
* Status: Status of the sale. (String)
* Fulfilment: Method of fulfilment. (String)
* Style: Style of the product. (String)
* SKU: Stock Keeping Unit. (String)
* ASIN: Amazon Standard Identification Number. (String)
* Courier Status: Status of the courier. (String)
* Qty: Quantity of the product. (Integer)
* Amount: Amount of the sale. (Float)
* B2B: Business to business sale. (Boolean)
* Currency: The currency used for the sale. (String)

In [12]:
# To identify the column names

df.columns

Index(['index', 'Order ID', 'Date', 'Status', 'Fulfilment', 'Sales Channel ',
       'ship-service-level', 'Style', 'SKU', 'Category', 'Size', 'ASIN',
       'Courier Status', 'Qty', 'currency', 'Amount', 'ship-city',
       'ship-state', 'ship-postal-code', 'ship-country', 'promotion-ids',
       'B2B', 'fulfilled-by'],
      dtype='object')

### Unique Values

Unique values are the distinct or different values present in a column,without any repetition
We use the function unique(),example

df['Status'].unique()

In [13]:
df['Fulfilment'].unique()

array(['Merchant', 'Amazon'], dtype=object)

### Count unique values

In [22]:
#Example for nunique()
df['Status'].nunique()

13

### Showing value counts

In [23]:
#Example for value counts
df['Status'].value_counts

<bound method IndexOpsMixin.value_counts of 0                            Cancelled
1         Shipped - Delivered to Buyer
2                              Shipped
3                            Cancelled
4                              Shipped
                      ...             
128970                         Shipped
128971                         Shipped
128972                         Shipped
128973                         Shipped
128974                         Shipped
Name: Status, Length: 128975, dtype: object>

## Statistical Summary

A statistical summary is a brief overview of the key numerical characteristics of a dataset
It helps to understand
* Distribution
* Cental tendency
* Spread of data

### Statistical Measures includes,
* count:Number of non null values
* mean:Average value
* std:Standard deviation(variability in the data)
* min:Minimum value
* max:Maximum value
* 25%,50%,75% Quartiles (Q1,median,Q3)

In pandas we use describe() function to display the statistical summary of the dataset

In [14]:
# Statistical summary

df.describe()

index                           Date            Qty  \
count  128975.000000                         128975  128975.000000   
mean    64487.000000  2022-05-12 11:49:27.951928576       0.904431   
min         0.000000            2022-03-31 00:00:00       0.000000   
25%     32243.500000            2022-04-20 00:00:00       1.000000   
50%     64487.000000            2022-05-10 00:00:00       1.000000   
75%     96730.500000            2022-06-04 00:00:00       1.000000   
max    128974.000000            2022-06-29 00:00:00      15.000000   
std     37232.019822                            NaN       0.313354   

              Amount  ship-postal-code  
count  121180.000000     128942.000000  
mean      648.561465     463966.236509  
min         0.000000     110001.000000  
25%       449.000000     382421.000000  
50%       605.000000     500033.000000  
75%       788.000000     600024.000000  
max      5584.000000     989898.000000  
std       281.211687     191476.764941

## Data Quality Checks

### Values:
* Missing data can affect statistical calculations and model performance.
* Need to identify which columns have missing values
* This help to plan appropriate handling strategies.


### Duplicate Rows

* Duplicate records can twist analysis results.
* They may indicate entry errors.
* Duplicates should be removed to ensure data integrity

In [ ]:
The isnull().sum() is used to count missing values and duplicated().sum() is used to count duplicate rows.

In [15]:
#Check for missing values

print("Missing values summary:")
df.isnull().sum()

Missing values summary:


index                     0
Order ID                  0
Date                      0
Status                    0
Fulfilment                0
Sales Channel             0
ship-service-level        0
Style                     0
SKU                       0
Category                  0
Size                      0
ASIN                      0
Courier Status         6872
Qty                       0
currency               7795
Amount                 7795
ship-city                33
ship-state               33
ship-postal-code         33
ship-country             33
promotion-ids         49153
B2B                       0
fulfilled-by          89698
dtype: int64

In [26]:
# To check for Duplicate values

df.duplicated()

0         False
1         False
2         False
3         False
4         False
          ...  
128970    False
128971    False
128972    False
128973    False
128974    False
Length: 128975, dtype: bool

There is no duplicate rows in this dataset.

### Datatype error:
* Date:Stored as object instead of datetime
* Format is also seems different.Instead of dd-mm-yyyy ,it is stored as mm-dd-yy.

In [13]:
df.head()

index             Order ID       Date                        Status  \
0      0  405-8078784-5731545 2022-04-30                     Cancelled   
1      1  171-9198151-1101146 2022-04-30  Shipped - Delivered to Buyer   
2      2  404-0687676-7273146 2022-04-30                       Shipped   
3      3  403-9615377-8133951 2022-04-30                     Cancelled   
4      4  407-1069790-7240320 2022-04-30                       Shipped   

  Fulfilment Sales Channel  ship-service-level    Style              SKU  \
0   Merchant      Amazon.in           Standard   SET389   SET389-KR-NP-S   
1   Merchant      Amazon.in           Standard  JNE3781  JNE3781-KR-XXXL   
2     Amazon      Amazon.in          Expedited  JNE3371    JNE3371-KR-XL   
3   Merchant      Amazon.in           Standard    J0341       J0341-DR-L   
4     Amazon      Amazon.in          Expedited  JNE3671  JNE3671-TU-XXXL   

        Category  ... Qty currency  Amount    ship-city   ship-state  \
0            Set  ...   0      INR  647.62       MUMBAI  MAHARASHTRA   
1          kurta  ...   1      INR  406.00    BENGALURU    KARNATAKA   
2          kurta  ...   1      INR  329.00  NAVI MUMBAI  MAHARASHTRA   
3  Western Dress  ...   0      INR  753.33   PUDUCHERRY   PUDUCHERRY   
4            Top  ...   1      INR  574.00      CHENNAI   TAMIL NADU   

   ship-postal-code ship-country  \
0          400081.0           IN   
1          560085.0           IN   
2          410210.0           IN   
3          605008.0           IN   
4          600073.0           IN   

                                       promotion-ids    B2B fulfilled-by  
0                                                NaN  False    Easy Ship  
1  Amazon PLCC Free-Financing Universal Merchant ...  False    Easy Ship  
2       IN Core Free Shipping 2015/04/08 23-48-5-108   True          NaN  
3                                                NaN  False    Easy Ship  
4                                                NaN  False          NaN  

[5 rows x 23 columns]

In [51]:
df.dtypes

index                          int64
Order ID                      object
Date                  datetime64[ns]
Status                        object
Fulfilment                    object
Sales Channel                 object
ship-service-level            object
Style                         object
SKU                           object
Category                      object
Size                          object
ASIN                          object
Courier Status                object
Qty                            int64
currency                      object
Amount                       float64
ship-city                     object
ship-state                    object
ship-postal-code              object
ship-country                  object
promotion-ids                 object
B2B                             bool
fulfilled-by                  object
dtype: object

## Handling of Missing Values
* Missing values occur when data is not recorded or unavailable
* It can affect data analysis and lead to incorrect results.
* First step is to identify missing values using isnull() or isna()
* Numeric missing values can be filled by mean, median,mode
* Categorical values can be filled with most frequent values
* Rows and columns with too many missing values may be removed
* Proper handling improves data accuracy and reliability

### Missing values in this dataset

* Courier Status:6872
* Currency:7795
* Amount:7795
* Ship-city:33
* Ship-state:33
* Ship-postal-code:33
* Ship-country-33
* Promotion-ids:49153
* fulfilled-by:89698

## Handling missing values includes,
### Removal of Missing Data
* Rows or columns containing missing values are deleted.
* Used when missing values are very few and do not affect analysis

### Imputation (Mean,Median,Mode)
Missing numerical values are replaced with:
* Mean-When data is normally distributed
* Median-When data is skewed.
Missing categorical values are replaced with:
* Mode(Most frequent value)

### Constant Value Imputation
* Missing values are replaced with a fixed value such as "unknown" or 0.

### Forward Fill and Backward Fill
* Commonly used in time-series data.
* Fills missing values using previous or next available value

### Interpolation
* Estimates missing numerical values based on nearby data points

### Handling of missing values in the dataset

In [16]:
# Handling of missing value in Courier Statue column
df['Courier Status']=df['Courier Status'].fillna('Unknown')

In [17]:
# Handling of missing values in Currency column
df['currency']=df['currency'].fillna('INR')

In [18]:
# Handling of missing values in Amount column
df['Amount']=df['Amount'].fillna(df['Amount'].mean())

In [19]:
# Handling of missing values in ship-city,ship-state,ship-postal-code,ship-country

ship_cols=['ship-city','ship-state','ship-postal-code','ship-country']

df[ship_cols]=df[ship_cols].fillna("Unknown")

In [20]:
# Handling of missing values in promotion-ids column
df['promotion-ids']=df['promotion-ids'].fillna('Unknown')

In [21]:
# Handling of missing values in fulfilled-by column
df['fulfilled-by']=df['fulfilled-by'].fillna('Unknown')

In [22]:
df.dtypes

index                          int64
Order ID                      object
Date                  datetime64[ns]
Status                        object
Fulfilment                    object
Sales Channel                 object
ship-service-level            object
Style                         object
SKU                           object
Category                      object
Size                          object
ASIN                          object
Courier Status                object
Qty                            int64
currency                      object
Amount                       float64
ship-city                     object
ship-state                    object
ship-postal-code              object
ship-country                  object
promotion-ids                 object
B2B                             bool
fulfilled-by                  object
dtype: object

In [23]:
df.isnull().sum()

index                 0
Order ID              0
Date                  0
Status                0
Fulfilment            0
Sales Channel         0
ship-service-level    0
Style                 0
SKU                   0
Category              0
Size                  0
ASIN                  0
Courier Status        0
Qty                   0
currency              0
Amount                0
ship-city             0
ship-state            0
ship-postal-code      0
ship-country          0
promotion-ids         0
B2B                   0
fulfilled-by          0
dtype: int64

### Result

All the missing values handled successfully.


## Creating Derived Columns

As a part of the data preprocessing stage,new derived columns can be created from the existing dataset to support better analysis and meaningful insights.These derived attributes help in understanding sales trends, customer behaviour and fulfillment performance more effectively

This derived column helps in analyzing:

* Overall revenue generated

* Sales trends over time

* Revenue contribution by product category

* Regional revenue distribution

Creating this column enables accurate financial and performance analysis of Amazon e-commerce sales data.

### 1 Order Month

In [24]:
#Creation of Derived column Order Month
df['Date']=pd.to_datetime(df['Date'])
df['Order_Month']=df['Date'].dt.month_name()


In [10]:
df.columns #checking whether new column is created or not.

Index(['index', 'Order ID', 'Date', 'Status', 'Fulfilment', 'Sales Channel ',
       'ship-service-level', 'Style', 'SKU', 'Category', 'Size', 'ASIN',
       'Courier Status', 'Qty', 'currency', 'Amount', 'ship-city',
       'ship-state', 'ship-postal-code', 'ship-country', 'promotion-ids',
       'B2B', 'fulfilled-by', 'Order_Month'],
      dtype='object')

A new column was created to extract the month from the order date.This helps in analyzing monthly sales trends and seasonality patterns.

### 2 Order Year

In [25]:
#Creation of Derived column Order year
df['Order_year']=df['Date'].dt.year

The year was extracted from the order date to enable year-wise comparison of sales performance.

### 3 Order Day of week

In [26]:
#Derived column Order day
df['Order_Day'] = df['Date'].dt.day_name()




This column identifies the day on which the order was placed, helping to analyze peak ordering days.

### 4 Order Type (B2B / B2C)

In [27]:
#Derived column Order type
df['Order_Type'] = df['B2B'].apply(lambda x: 'B2B' if x==True else 'B2C')


Orders were classified into Business-to-Business (B2B) and Business-to-Customer (B2C) to understand customer segmentation.

### 5 Fulfillment Type

In [28]:
#Derived column Fulfillment Type
df['Fulfillment_Type'] = df['fulfilled-by'].apply(lambda x: 'Amazon' if 'Amazon' in str(x) else 'Seller')

This column categorizes orders based on whether they were fulfilled by Amazon or by third-party sellers, allowing analysis of fulfillment efficiency.

### 6 Order Success Status

In [29]:
#Derived column order success status
df['Order_Success'] = df['Status'].apply(lambda x: 'Successful' if x=='Delivered' else 'Not Successful')

Orders were grouped into successful and unsuccessful deliveries based on the order status to evaluate fulfillment performance.

### 7 Region

In [36]:
north = ['DELHI','HARYANA','PUNJAB','HIMACHAL PRADESH','UTTARAKHAND',
         'UTTAR PRADESH','JAMMU & KASHMIR','LADAKH','CHANDIGARH']

south = ['TAMIL NADU','KERALA','KARNATAKA',
         'ANDHRA PRADESH','TELANGANA','PUDUCHERRY']

east = ['WEST BENGAL','ODISHA','BIHAR','JHARKHAND','ASSAM','MEGHALAYA',
        'MANIPUR','MIZORAM','NAGALAND','TRIPURA','ARUNACHAL PRADESH',
        'SIKKIM','ANDAMAN & NICOBAR']

west = ['MAHARASHTRA','GUJARAT','RAJASTHAN','GOA',
        'DADRA AND NAGAR','DAMAN & DIU','MADHYA PRADESH','CHHATTISGARH']

In [39]:
region_list = []

for state in df['ship-state']:
    state = state.upper().strip()
    
    if state in north:
        region_list.append('North')
    elif state in south:
        region_list.append('South')
    elif state in east:
        region_list.append('East')
    elif state in west:
        region_list.append('West')
    else:
        region_list.append('Unknown')

df['Region'] = region_list


A region column was created from the shipping state to perform regional sales analysis

### 8 Postal code prefix

In [30]:
# Derived column Postal area
df['Postal_Area'] = df['ship-postal-code'].astype(str).str[:3]

### 9 Total Sales Value

In [31]:
#Derived column total sales value
df['Total_Sales'] = df['Qty'] * df['Amount']

In [33]:
df.columns #All columns can be seen,,Derived columns are created successfully.

Index(['index', 'Order ID', 'Date', 'Status', 'Fulfilment', 'Sales Channel ',
       'ship-service-level', 'Style', 'SKU', 'Category', 'Size', 'ASIN',
       'Courier Status', 'Qty', 'currency', 'Amount', 'ship-city',
       'ship-state', 'ship-postal-code', 'ship-country', 'promotion-ids',
       'B2B', 'fulfilled-by', 'Order_Month', 'Order_year', 'Order_Day',
       'Order_Type', 'Fulfillment_Type', 'Order_Success', 'Postal_Area',
       'Total_Sales', 'Promo_Used'],
      dtype='object')

### 5. Promotion Flag (Yes/No)

In [32]:
#derived column promotion used
df['Promo_Used'] = df['promotion-ids'].apply(lambda x: 'Yes' if x=='Yes' else 'No')

In [33]:
df.isnull().sum()

index                 0
Order ID              0
Date                  0
Status                0
Fulfilment            0
Sales Channel         0
ship-service-level    0
Style                 0
SKU                   0
Category              0
Size                  0
ASIN                  0
Courier Status        0
Qty                   0
currency              0
Amount                0
ship-city             0
ship-state            0
ship-postal-code      0
ship-country          0
promotion-ids         0
B2B                   0
fulfilled-by          0
Order_Month           0
Order_year            0
Order_Day             0
Order_Type            0
Fulfillment_Type      0
Order_Success         0
Region                0
Postal_Area           0
Total_Sales           0
Promo_Used            0
dtype: int64

## Filtering

Filtering is used to select specific subsets of data based on conditions. For example, orders with a particular status, category, or promotion type were filtered for focused analysis.

Examples of filtering operations include:

* Selecting only "shipped-out for delivery "orders

* Filtering orders where promotions were applied

* Separating B2B and B2C orders

* Analyzing specific product categories

### Filtering only "Shipped-out for delivery"

In [34]:
# 1. Filter only "shipped-out for delivery" orders
Shipped_delivery = df[df['Status'] == 'Shipped - Out for Delivery']
Shipped_delivery .head()

index             Order ID       Date                      Status  \
15631  15631  406-6849237-1026769 2022-04-21  Shipped - Out for Delivery   
69240  69240  405-9887214-3578748 2022-05-15  Shipped - Out for Delivery   
95520  95520  405-5743429-6628355 2022-06-25  Shipped - Out for Delivery   
96755  96755  405-1365097-0132364 2022-06-24  Shipped - Out for Delivery   
96911  96911  408-3653655-5699510 2022-06-24  Shipped - Out for Delivery   

      Fulfilment Sales Channel  ship-service-level    Style              SKU  \
15631   Merchant      Amazon.in           Standard   SET398   SET398-KR-PP-M   
69240   Merchant      Amazon.in           Standard   SET156   SET156-KR-NP-M   
95520   Merchant      Amazon.in           Standard   SET179  SET179-KR-PP-XS   
96755   Merchant      Amazon.in           Standard  JNE3661    JNE3661-TP-XL   
96911   Merchant      Amazon.in           Standard    J0230      J0230-SKD-L   

      Category  ... Order_Month Order_year Order_Day  Order_Type  \
15631      Set  ...       April       2022  Thursday         B2C   
69240      Set  ...         May       2022    Sunday         B2C   
95520      Set  ...        June       2022  Saturday         B2C   
96755      Top  ...        June       2022    Friday         B2C   
96911      Set  ...        June       2022    Friday         B2C   

      Fulfillment_Type   Order_Success Region Postal_Area Total_Sales  \
15631           Seller  Not Successful  Other         737      1115.0   
69240           Seller  Not Successful  Other         185       680.0   
95520           Seller  Not Successful  Other         410       499.0   
96755           Seller  Not Successful  Other         411       301.0   
96911           Seller  Not Successful  Other         250      1163.0   

      Promo_Used  
15631         No  
69240         No  
95520         No  
96755         No  
96911         No  

[5 rows x 33 columns]

### Filter orders with promotion is not used

In [35]:
# 2. Filter orders with Promotion is not used
promo_orders = df[df['Promo_Used'] == 'No']
promo_orders.head()

index             Order ID       Date                        Status  \
0      0  405-8078784-5731545 2022-04-30                     Cancelled   
1      1  171-9198151-1101146 2022-04-30  Shipped - Delivered to Buyer   
2      2  404-0687676-7273146 2022-04-30                       Shipped   
3      3  403-9615377-8133951 2022-04-30                     Cancelled   
4      4  407-1069790-7240320 2022-04-30                       Shipped   

  Fulfilment Sales Channel  ship-service-level    Style              SKU  \
0   Merchant      Amazon.in           Standard   SET389   SET389-KR-NP-S   
1   Merchant      Amazon.in           Standard  JNE3781  JNE3781-KR-XXXL   
2     Amazon      Amazon.in          Expedited  JNE3371    JNE3371-KR-XL   
3   Merchant      Amazon.in           Standard    J0341       J0341-DR-L   
4     Amazon      Amazon.in          Expedited  JNE3671  JNE3671-TU-XXXL   

        Category  ... Order_Month Order_year Order_Day  Order_Type  \
0            Set  ...       April       2022  Saturday         B2C   
1          kurta  ...       April       2022  Saturday         B2C   
2          kurta  ...       April       2022  Saturday         B2B   
3  Western Dress  ...       April       2022  Saturday         B2C   
4            Top  ...       April       2022  Saturday         B2C   

  Fulfillment_Type   Order_Success Region Postal_Area Total_Sales Promo_Used  
0           Seller  Not Successful  Other         400         0.0         No  
1           Seller  Not Successful  Other         560       406.0         No  
2           Seller  Not Successful  Other         410       329.0         No  
3           Seller  Not Successful  Other         605         0.0         No  
4           Seller  Not Successful  Other         600       574.0         No  

[5 rows x 33 columns]

### Filter B2B orders

In [66]:
# 3. Filter B2B orders
b2b_orders = df[df['B2B'] == True]


### Filter specific category

In [67]:
#filter specific category kurta
kurta=df[df['Category']=='Kurta']



## Aggregation

Aggregation was performed to summarize data and compute overall statistics. Grouping operations were applied to calculate total and average sales across different dimensions.

Examples of aggregation operations include:

* Total sales by product category

* Total sales by shipping state

* Monthly sales trends

* Average order value

In [68]:
# 1. Total sales by Category
category_sales = df.groupby('Category')['Total_Sales'].sum()
print("Total sales by category",category_sales)

Total sales by category Category
Blouse           4.535817e+05
Bottom           1.435186e+05
Dupatta          9.150000e+02
Ethnic Dress     7.629490e+05
Saree            1.257670e+05
Set              3.797594e+07
Top              5.245525e+06
Western Dress    1.071053e+07
kurta            2.072399e+07
Name: Total_Sales, dtype: float64


In [69]:
# 2. Total sales by State
state_sales = df.groupby('ship-state')['Total_Sales'].sum()
print("Total sales by state",state_sales)

Total sales by state ship-state
ANDAMAN & NICOBAR      149939.0
ANDHRA PRADESH        3109687.0
APO                         0.0
AR                        493.0
ARUNACHAL PRADESH       92535.0
                        ...    
goa                      2467.0
orissa                    597.0
punjab                   8622.0
rajasthan                3946.0
rajsthan                  521.0
Name: Total_Sales, Length: 70, dtype: float64


In [70]:
# 3. Monthly sales trend
monthly_sales = df.groupby('Order_Month')['Total_Sales'].sum()
print("Total sales by month",monthly_sales)

Total sales by month Order_Month
April    7.614272e+07
Name: Total_Sales, dtype: float64


In [155]:
# 4. Average order value
avg_order_value = df['Total_Sales'].mean()
print("Average order value",avg_order_value)

Average order value 590.3680229859698


In [71]:
# Total sales of shipped-out for delivery
out_delivered_sales = df[df['Status']=='Shipped - Out for Delivery'].groupby('Category')['Total_Sales'].sum()
print("Total sales of shipped-out for delivery",out_delivered_sales)

Total sales of shipped-out for delivery Category
Set              17254.0
Top               2518.0
Western Dress     4590.0
kurta             2609.0
Name: Total_Sales, dtype: float64


### Purpose

Filtering helps in narrowing down the dataset to relevant records, while aggregation helps in summarizing large volumes of data into meaningful metrics. Together, these techniques support trend analysis, performance evaluation, and data-driven decision-making in the project.

In [42]:
#Saving cleaned dataset as csv file
df.to_csv("cleaned.csv",index=False)
print("Cleaned dataset saved successfully")

Cleaned dataset saved successfully
